src/null-safety/index.md

```diff
@@ -1,39 +1,257 @@
 ---
 title: Null safety
-description: Watch this space for news about null safety
-toc: false
+description: Information about Dart's upcoming null safety feature
 ---
 
-Watch this page for updates about null safety.
-In the meantime, check out these resources:
+Sound null safety is coming to the Dart language!
+When you opt into null safety,
+types in your code are non-nullable by default, meaning that
+values can’t be null _unless you say they can be._
+With null safety, your **runtime** null-dereference errors
+turn into **edit-time** analysis errors.
+The compiler can optimize away internal null checks,
+enabling apps to be faster, smaller, and more reliable.
 
-* [Null-safe playground: nullsafety.dartpad.dev][nullsafety.dartpad.dev]
+You can practice using null safety in the web app
+[DartPad with Null Safety.][nullsafety.dartpad.dev]
+
+{{ site.alert.important }}
+  **Don't start migrating large projects yet.**
+  Do test the feature and [give us feedback.][]
+{{ site.alert.end }}
+
+[give us feedback.]: https://github.com/dart-lang/sdk/issues/new?title=Null%20safety%20feedback:%20[issue%20summary]&labels=NNBD&body=Describe%20the%20issue%20or%20potential%20improvement%20in%20detail%20here
+
+If you've used Kotlin, TypeScript, or C#,
+then some of the null-safety syntax might look familiar.
+That's by design: the Dart language aims to be unsurprising.
+
+
+## Creating and using variables (?, !, late)
+
+Use `?`, `!`, and `late` to tweak how the Dart analyzer treats your code.
```

Comment:
'tweak' sounds a bit odd here; these are not 'fine adjustments'. And it's not just for the analyzer, it's a fundamental type change which also effects compilers, etc.

Reply:
Agree. Something like "... to tell the compiler how your code handles null values."

Reply:
Or maybe something like:

Use `?`, `!`, and `late` to specify the nullability of a variable. Because null safety in Dart is sound, this effects the behaviour of the Dart analyzer, and the Dart compilers & runtimes.

Reply:
I'm using mit's first sentence but not the second. (If we want something like the second sentence, it should go in the intro to this page.)

Reply:
I really like your idea of expanding on this in the intro. Maybe something like:

## Sound null safety

When you use null safety to specify the nullability of your variables, the Dart analyzer will enforce that your code is safe. For example that you don't read a nullable variable without checking for null. You will see potential issues listed next to other warnings & errors from the Dart analyzer.

[screenshot of analyzer output?]

Further, unlike some other languages, null safety in Dart is sound: This means that Dart is 100% sure that a variable isn't null if the nullability specification says so. As a result, the Dart compilers & runtimes can produce optimized code which is smaller and faster.

Reply:
This had some overlap with what we already had, so I've modified it heavily. ptal.

Reply:
From a marketing PoV I'd really like to have the 'Sound null safety' heading. WDYT?